### ISSUE TITLE
Support for Node.js

### ISSUE THREAD

**Issue by maarten-22** [usnistgov/jsfive#19]
Are there any plans to make this available/work in Node.js?
I could only find two hdf5 packages for JavaScript; this one and https://github.com/HDF-NI/hdf5.node. However hdf5.node does not work on Windows currently.

**Comment by bmaranville** [comment]
You can use it right now with a recent version of Node.js (v 14+?), but only with a local install...
Even from the REPL... if you clone this repo you can do "node --experimental-repl-await" and then from the jsfive directory you can load it in node with "const hdf5 = await import("./index.js");"

If you want it packaged as a node module, so it can be more easily loaded with npm install etc. I can look into adding a node build.

**Comment by bmaranville** [comment]
I added a node build - you can now install it with npm by e.g.
npm install https://github.com/usnistgov/jsfive/archive/refs/tags/v0.3.1.tar.gz
or 
npm install git+https://github.com/usnistgov/jsfive.git

After that you can use ```const hdf5 = require("jsfive");``` in your nodejs code.  If this works for you, I will close this ticket.

**Comment by maarten-22** [comment]
Awesome! Much better already, but I'm struggling to get the code to work. When using the example from readme with node-fetch, I get `TypeError: hdf5.File is not a constructor`. 
And `console.log(hdf5)` returns `{ dataloader_callbacks: [], Group: [class Si], Dataset: [class Mi] }`.
How is this module used?

**Comment by bmaranville** [comment]
You're right... the bundle is missing "File" in the exports.  I don't have any idea why Parcel2 included Group and Dataset but not File, though they are all imported/exported the same way in index.js

**Comment by bmaranville** [comment]
Ok, I learned a little bit more about packaging for node and now there is a working build ready for npm install (I tried it and it seems to work fine).  The version number is bumped to 0.3.2

There is sample usage information at the end of the README on the main github page for jsfive now: 

```
$ node
Welcome to Node.js v14.17.4.
Type ".help" for more information.
> const hdf5 = require("jsfive")
undefined
> var fs = require("fs");
undefined
> var ab = fs.readFileSync("/home/brian/Downloads/sans4490.nxs.ngv");
undefined
> var f = new hdf5.File(ab.buffer);
undefined
> f.keys
[ 'entry' ]
> f.get("entry").attrs
{ NX_class: 'NXentry' }
> 
```

**Comment by bmaranville** [comment]
I published version 0.3.3 to npmjs so you should be able to do "npm install jsfive" now.

**Comment by maarten-22** [comment]
Ah you're the best! Thanks a lot. Did a quick test and seems to work fine.
